### PR TITLE
Fix rectangle resize with rotation

### LINKED ISSUE #2742: Drawing: rotation lost when resizing a rectangle
I don't know if it is a current limitation or a new issue. 
When drawing a rectangle, then rotate it (right click action), the resize action will reset the rotation.

### PR DESCRIPTION
**Pull request by alexbrault**
Fixes #2742

When starting to drag a rectangle, the sides vectors are precalculated and used for the duration of the resize operation instead of as the rectangle is resized. If one of the vectors is invalid (i.e. the situation that caused the rectangles to turn into triangles), we can replace it with a vector perpendicular to the other, valid vector. 
In the very unlikely event of both vectors being invalid (which requires all 4 corners of the rectangle to be at the exact same point when the resize starts), the algorithm still maintains a rectangle shape.

### CODE REVIEW

**Comment by gberaudo** [comment]
Hi @alexbrault, could you please rebase?

File: src/ol-ext/interaction/modifyrectangle.js [hunk 1]
```diff
@@ -191,11 +191,81 @@ ngeo.interaction.ModifyRectangle.prototype.willModifyFeatures_ = function(evt) {
     this.modified_ = true;
     this.dispatchEvent(new ol.interaction.Modify.Event(
       ol.interaction.ModifyEventType.MODIFYSTART, this.features_, evt));
+    this.params_ = this.initializeParams_();
```
Comment (gberaudo) [inline src/ol-ext/interaction/modifyrectangle.js]: Why don't you put the initialization in the `handleDown_` listener?

Comment (alexbrault) [inline src/ol-ext/interaction/modifyrectangle.js]: The idea was to delay the vector calculation until the resizing actually starts so we don't calculate the vectors for nothing each time the user clicks on a feature.

Comment (gberaudo) [inline src/ol-ext/interaction/modifyrectangle.js]: ok

**Comment by gberaudo** [comment]
@alexbrault, I ran the test and it looked to work well. There are some compiler errors in Travis. Can you come up with a fix? Then I think we can merge.

**Comment by sbrunner** [comment]
Thanks @alexbrault :-)

**Pull request merged**